### PR TITLE
store implementation

### PR DESCRIPTION
**Pull request by ArunRajBTech**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div class="container" style="width: 700px;margin-top: 5rem;">
```
Comment (mbelraka) [inline src/app/app.component.html:1]: avoid inline styling

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-<div *ngFor="let data of sampleData">
+<div *ngFor="let data of sampleData | async">
```
Comment (mbelraka) [inline src/app/app.component.html:2]: use descriptive name, what are these data?

Comment (mbelraka) [reply src/app/app.component.html:2]: Please mark observables with $ at the end, e.g. sampleData$

File: src/app/app.component.ts [hunk 3]
```diff
@@ -8,12 +13,11 @@ import {TestService} from './test.service';
 })
 export class AppComponent implements OnInit {
   title = 'samplecardexample';
+  sampleData = this.store.pipe(select(itemData));
```
Comment (mbelraka) [inline src/app/app.component.ts:16]: sampleData$ - descriptive name

File: src/app/app.component.ts [hunk 4]
```diff
@@ -1,5 +1,10 @@
 import { Component, OnInit } from '@angular/core';
 import {TestService} from './test.service';
+import{select, Store} from '@ngrx/store'
+import {Item} from '../models/item.model'
+import { Observable } from 'rxjs';
+import { invokeItemAPI } from 'src/store/item.action';
+import { itemData } from 'src/store/item.selector';
```
Comment (mbelraka) [inline src/app/app.component.ts:7]: group imports together: angular - libraries - internal imports

File: src/app/app.component.ts [hunk 5]
```diff
@@ -8,12 +13,11 @@ import {TestService} from './test.service';
 })
 export class AppComponent implements OnInit {
   title = 'samplecardexample';
+  sampleData = this.store.pipe(select(itemData));
 
-  public sampleData:any[]=[];
-  constructor(private testService: TestService) { }
+  constructor(private store: Store<{item:Item[]}>) { }
```
Comment (mbelraka) [inline src/app/app.component.ts:18]: private readonly store: Store<{item:Item[]}>

please use explicit public, private or protected

File: src/app/app.component.ts [hunk 6]
```diff
@@ -8,12 +13,11 @@ import {TestService} from './test.service';
...
 
   ngOnInit(){
```
Comment (mbelraka) [inline src/app/app.component.ts:20]: explicit modifiers, public, private or protected
explicit return types, return type or void

File: src/app/app.component.ts (hunk 5 above)
Comment (mbelraka) [reply src/app/app.component.ts:18]: private variables should include _ at the start, e.g. _store

File: src/app/test.service.ts [hunk 7]
```diff
@@ -10,8 +12,7 @@ export class TestService {
   constructor(private http: HttpClient) { }
 
   public getData() {
-    return this.http.get<any>('/assets/files.json')
-      .toPromise()
-      .then(res => <any[]>res.data);
+    return this.http.get('/assets/files.json')
+      .pipe((map((data)=>data)));
```
Comment (mbelraka) [inline src/app/test.service.ts:16]: map(data => ObjectMapper.deserializeArray(type, data))

File: src/models/item.model.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export interface Item{
+id:string,
+title:string,
+content:string,
+image:string
```
Comment (mbelraka) [inline src/models/item.model.ts:5]: descriptive name- should be class

File: src/store/app.state.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import {Item} from '../models/item.model'
+
+export interface AppState{
```
Comment (mbelraka) [inline src/store/app.state.ts:3]: should be a class

File: src/store/item.action.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import {createAction, props} from'@ngrx/store';
+import {Item} from '../models/item.model'
+
+export const retrieveItem=createAction(
+    '[Item API] API success',
+    props<{allItem:Item[]}>()
+);
+
+export const invokeItemAPI=createAction(
+    '[Item API] Invoke API'
```
Comment (mbelraka) [inline src/store/item.action.ts:10]: store strings outside classes

File: src/store/item.effects.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { TestService } from 'src/app/test.service';
+import {map, mergeMap} from 'rxjs/operators'
+
+@Injectable()
```
Comment (mbelraka) [inline src/store/item.effects.ts:6]: providedIn: root

File: src/store/item.effects.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+export class ItemEffects{
+    constructor(private $action:Actions,
+        private itemService:TestService){
+
```
Comment (mbelraka) [inline src/store/item.effects.ts:10]: delete the emthod

File: src/store/item.effects.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        $loadItem=createEffect(()=>
+            this.$action.pipe(
+               ofType('[Item API] Invoke API'),
+               mergeMap(()=>
+                 this.itemService.getData()
+                .pipe(
+                 map((data)=>({type:'[Item API] API success',allItem:data}))
```
Comment (mbelraka) [inline src/store/item.effects.ts:19]: store strings outside classes

File: src/store/item.reducer.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+export const initialState:Item[]=[];
+
+const _itemReducer=createReducer(
+    initialState,
+on(retrieveItem,(state,{allItem})=>{
+    return [...allItem]
+})
+);
+
+export function itemReducer(state:any,actions:any){
```
Comment (mbelraka) [inline src/store/item.reducer.ts:14]: don't use any, use specific types

**Review by mbelraka — CHANGES_REQUESTED** [review]

File: src/app/app.component.html (hunk 1 above)
Comment (ArunRajBTech) [reply src/app/app.component.html:1]: removed

File: src/app/app.component.html (hunk 2 above)
Comment (ArunRajBTech) [reply src/app/app.component.html:2]: Updated

File: src/app/app.component.ts (hunk 3 above)
Comment (ArunRajBTech) [reply src/app/app.component.ts:16]: Updated

File: src/app/app.component.ts (hunk 4 above)
Comment (ArunRajBTech) [reply src/app/app.component.ts:7]: Updated

File: src/app/app.component.ts (hunk 6 above)
Comment (ArunRajBTech) [reply src/app/app.component.ts:20]: Couldn't understand. Please explain bit

File: src/app/test.service.ts (hunk 7 above)
Comment (ArunRajBTech) [reply src/app/test.service.ts:16]: Updated the code

File: src/models/item.model.ts (hunk 8 above)
Comment (ArunRajBTech) [reply src/models/item.model.ts:5]: Updated

File: src/app/app.component.ts (hunk 6 above)
Comment (mbelraka) [reply src/app/app.component.ts:20]: public ngOnInit(): void {

}

File: src/app/app.component.ts (hunk 5 above)
Comment (ArunRajBTech) [reply src/app/app.component.ts:18]: updated

File: src/app/app.component.ts (hunk 6 above)
Comment (ArunRajBTech) [reply src/app/app.component.ts:20]: Updated

File: src/store/app.state.ts (hunk 9 above)
Comment (ArunRajBTech) [reply src/store/app.state.ts:3]: updated

File: src/store/item.action.ts (hunk 10 above)
Comment (ArunRajBTech) [reply src/store/item.action.ts:10]: updated

File: src/store/item.effects.ts (hunk 11 above)
Comment (ArunRajBTech) [reply src/store/item.effects.ts:6]: updated

File: src/store/item.effects.ts (hunk 12 above)
Comment (ArunRajBTech) [reply src/store/item.effects.ts:10]: Do I need to delete the entire method?

File: src/store/item.effects.ts (hunk 13 above)
Comment (ArunRajBTech) [reply src/store/item.effects.ts:19]: updated

File: src/store/item.effects.ts (hunk 12 above)
Comment (mbelraka) [reply src/store/item.effects.ts:10]: delete any unused code

File: src/store/item.reducer.ts (hunk 14 above)
Comment (ArunRajBTech) [reply src/store/item.reducer.ts:14]: updated

File: src/store/item.effects.ts (hunk 12 above)
Comment (ArunRajBTech) [reply src/store/item.effects.ts:10]: its being used

**Pull request merged**